src/util/tracing/goblint_tracing.ml

```diff
@@ -38,9 +39,13 @@ let traceTag (sys : string) : Pretty.doc =
   let rec ind (i : int) : string = if (i <= 0) then "" else " " ^ (ind (i-1)) in
   (text ((ind !indent_level) ^ "%%% " ^ sys ^ ": "))
 
+let trace_mutex = GobMutex.create ()
```

Comment:
Tracing isn't the only output from Goblint. Why don't others (like `Logs` and `Messages`) also need locks?

Reply:
Those should happen at the end in postprocessing, which is not concurrent afaik.

Reply:
Logging with `Logs` can happen at any time I think, although it might just be that the transfer functions rarely use it (and prefer `Messages`, for a good reason), but I haven't checked.
`Messages` could be used with `warn_at` being `early`, it's just not the default.

Reply:
I would leave this for the later PRs that contain the actual solver implementations. Traces here are included as an example for handling the kinds of situations where the concurrent implementation differs from the sequential. 

Practically: We need a shared mutex for all kinds of inputs (because logging  and tracing can interfere as well). However, this was not an issue during development.